src/odemis/driver/piezomotor.py

```diff
@@ -592,18 +645,16 @@ def isMovingClosedLoop(self, axis):
         """
         :param axis: (int) axis number
         :returns: (bool) True if moving, False otherwise
+        :raise: (PMDError) exception if error is detected.
         """
-        _, d2, d3, _ = self.getStatus(axis)
-        # Check d2 (second status value) bit 2 (external limit)
-        if d2 & 0b100:
-            logging.debug(f"External limit reached on axis {axis}, current position is {self.position.value}")
-            raise PMDError(6, f"External limit reached on axis {axis}.")
+        status = self.getStatus(axis)
+        self._check_axis_error(axis, status)
 
         # Check d3 (third status value) bit 2 (targetLimit: position limit reached) and bit 0 (targetReached)
-        if not d3 & 0b010:  # closed loop not active, thus it is not moving in closed loop
+        if not status[2] & 2:  # closed loop not active, thus it is not moving in closed loop
             logging.debug(f"Closed loop not active, therefore not moving in closed loop on axis {axis}.")
             return False
-        elif d3 & 0b101:
+        elif status[2] & 5:
```

Comment:
Why changing this? If it's called "d3", then it's best to call it "d3", not "status[2]" (but I don't remember the doc, if the doc has a different name, then anything is fine). Also, binary values are much more readable than an int, for this sort of bit checks. So I think it's best to keep the values in `0b...` format.

Reply:
The other status checks here: https://github.com/nandishjpatel/odemis/blob/faeec8a1943d7310e1e71e1998bb078acb914403/src/odemis/driver/piezomotor.py#L510 have ints. Should I also convert those to binary?

In the tech manual, there is use of d1, d2, d3 and d4 will adhere to them

Reply:
Also d1, d2, d3, d4 are already converted to int by `getStatus`

Reply:
As long as the ints are a power of 2, the math is easy to do in the head. So I don't really mind (though I think that writing in binary is always clearer). When there are multiple bits set, binary is a lot faster to read.

Reply:
> Also d1, d2, d3, d4 are already converted to int by `getStatus`
Binary values are int. It's just a matter of how they are written 🙂